lib/wallet/src/Cardano/Api/Extra.hs

```diff
@@ -135,3 +137,16 @@ fromShelleyBasedScript era script = case era of
                 ScriptInEra PlutusScriptV2InConway $
                 PlutusScript PlutusScriptV2 $
                 PlutusScriptSerialised s
+
+-- Not exposed by cardano-api
+unbundleLedgerShelleyBasedProtocolParams
+  :: ShelleyBasedEra era
+  -> BundledProtocolParameters era
+  -> Ledger.PParams (ShelleyLedgerEra era)
+unbundleLedgerShelleyBasedProtocolParams = \case
+  ShelleyBasedEraShelley -> \(BundleAsShelleyBasedProtocolParameters _ _ lpp) -> lpp
+  ShelleyBasedEraAllegra -> \(BundleAsShelleyBasedProtocolParameters _ _ lpp) -> lpp
+  ShelleyBasedEraMary -> \(BundleAsShelleyBasedProtocolParameters _ _ lpp) -> lpp
+  ShelleyBasedEraAlonzo -> \(BundleAsShelleyBasedProtocolParameters _ _ lpp) -> lpp
+  ShelleyBasedEraBabbage -> \(BundleAsShelleyBasedProtocolParameters _ _ lpp) -> lpp
+  ShelleyBasedEraConway -> \(BundleAsShelleyBasedProtocolParameters _ _ lpp) -> lpp
```

Comment:
Hmm. I'm guessing these are partial pattern matches?

The docs seem to indicate that there are two constructors:
https://input-output-hk.github.io/cardano-node/cardano-api/lib/Cardano-Api-ProtocolParameters.html#t:BundledProtocolParameters

Namely:
- `BundleAsByronProtocolParameters`
- `BundleAsShelleyBasedProtocolParameters`

Reply:
Now I see this is just a copy-and-paste of this function:
https://input-output-hk.github.io/cardano-node/cardano-api/lib/src/Cardano.Api.ProtocolParameters.html#unbundleLedgerShelleyBasedProtocolParams

Reply:
> BundleAsByronProtocolParameters

Yes, I think the compiler is smart and treats `IsShelleyBasedEra era => era ~ ByronEra` as impossible

Reply:
I wonder why this function isn't exposed by `cardano-api` 🤔